narwhals/_dask/expr.py

```diff
@@ -209,13 +210,57 @@ def __rtruediv__(self, other: Any) -> Self:
         ).alias("literal")
 
     def __floordiv__(self, other: Any) -> Self:
-        return self._with_callable(
-            lambda expr, other: expr.__floordiv__(other), "__floordiv__", other=other
+        def _floordiv(
+            df: DaskLazyFrame, series: dx.Series, other: dx.Series | Any
+        ) -> dx.Series:
+            series, other = align_series_full_broadcast(df, series, other)
+            return (series.__floordiv__(other)).where(other != 0, None)
+
+        def func(df: DaskLazyFrame) -> list[dx.Series]:
+            if isinstance(other, type(self)):
+                if len(other_ := other(df)) > 1:  # pragma: no cover
+                    msg = "Expected expression with single output, found multiple"
+                    raise ValueError(msg)
+                other_series = other_[0]
+            else:
+                other_series = other
+
+            return [_floordiv(df, series, other_series) for series in self(df)]
+
+        return self.__class__(
+            func,
+            depth=self._depth + 1,
+            function_name=self._function_name + "->__floordiv__",
+            evaluate_output_names=self._evaluate_output_names,
+            alias_output_names=self._alias_output_names,
+            version=self._version,
         )
 
     def __rfloordiv__(self, other: Any) -> Self:
-        return self._with_callable(
-            lambda expr, other: other // expr, "__rfloordiv__", other=other
+        def _rfloordiv(
+            df: DaskLazyFrame, series: dx.Series, other: dx.Series | Any
+        ) -> dx.Series:
+            series, other = align_series_full_broadcast(df, series, other)
+            return (other.__floordiv__(series)).where(series != 0, None)
+
+        def func(df: DaskLazyFrame) -> list[dx.Series]:
+            if isinstance(other, type(self)):
+                if len(other_ := other(df)) > 1:  # pragma: no cover
+                    msg = "Expected expression with single output, found multiple"
+                    raise ValueError(msg)
+                other_series = other_[0]
```

Comment:
Is this case even possible? It shouldn't be, right? As in if other is also a DaskExpr, then it would trigger `other.__floordiv__` ops.